src/Tasks/AssignTargetPath.cs

```diff
@@ -71,13 +71,20 @@ public override bool Execute()
 
                 for (int i = 0; i < Files.Length; ++i)
                 {
-                    string link = Files[i].GetMetadata(ItemMetadataNames.link);
                     AssignedFiles[i] = new TaskItem(Files[i]);
+                    string targetPath = Files[i].GetMetadata(ItemMetadataNames.targetPath);
 
-                    // If file has a link, use that.
-                    string targetPath = link;
+                    // If TargetPath is already set, copy it over.
+                    // https://github.com/dotnet/msbuild/issues/2795
+                    if (!string.IsNullOrEmpty(targetPath))
+                    {
+                        AssignedFiles[i].SetMetadata(ItemMetadataNames.targetPath, EscapingUtilities.Escape(targetPath));
```

Comment:
Consider using some new variable ("targetPathOverride"? "targetPath2"? Choose a good name) to allow users to provide a custom targetPath while maintaining that this task can be called twice without adverse consequences. (Escaping seems necessary as part of what TaskItem requires for all its metadata.)

Reply:
Going with `TargetPathOverride` for clarity.